models/classes/media/LocalItemSource.php

```diff
@@ -314,4 +267,67 @@ protected function getItemDirectory()
         }
         return $this->itemService->getItemDirectory($this->item, $this->lang);
     }
+
+    private function searchDirectories(
+        string $parentLink,
+        array $acceptableMime,
+        int $depth,
+        int $childrenLimit
+    ): array {
+        if (!tao_helpers_File::securityCheck($parentLink)) {
+            throw new common_exception_Error(__('Your path contains error'));
+        }
+
+        $label = rtrim($parentLink, '/');
+        if (strrpos($parentLink, '/') !== false && substr($parentLink, -1) !== '/') {
+            $label = substr($parentLink, strrpos($parentLink, '/') + 1);
+            $parentLink = $parentLink . '/';
+        }
+
+        if (in_array($parentLink, ['', '/'])) {
+            $label = $this->getItem()->getLabel();
+            $parentLink = '/';
+        }
+
+        $data = [
+            'path' => $parentLink,
+            'label' => $label,
+            'childrenLimit' => $childrenLimit,
+        ];
+
+        if ($depth <= 0) {
+            $data['parent'] = $parentLink;
+            return $data;
+        }
+
+        $children = [];
+
+        /** @var Directory $directory */
+        $itemDirectory = $this->getItemDirectory();
+        if ($parentLink != '/') {
+            $directory = $itemDirectory->getDirectory($parentLink);
+        } else {
+            $directory = $itemDirectory;
+        }
+
+        $iterator = $directory->getFlyIterator();
+        $total = 0;
+        foreach ($iterator as $content) {
+            if ($content instanceof Directory) {
+                $children[] = $this->searchDirectories($itemDirectory->getRelPath($content), $acceptableMime,
+                    $depth - 1, $childrenLimit);
+            } else {
+                $fileInfo = $this->getInfoFromFile($content);
+                if (empty($acceptableMime) || in_array($fileInfo['mime'], $acceptableMime)) {
+                    $children[] = $fileInfo;
+                    $total++;
+                }
+            }
```

Comment:
```suggestion
            if ($content instanceof Directory) {
                $children[] = $this->searchDirectories($itemDirectory->getRelPath($content), $acceptableMime,
                    $depth - 1, $childrenLimit);
            
                $total++;
                        
                continue; 
            }
                
            $fileInfo = $this->getInfoFromFile($content);
            
            if (empty($acceptableMime) || in_array($fileInfo['mime'], $acceptableMime)) {
                $children[] = $fileInfo;
                
                $total++;
            }
```

Reduce nested blocks / complexity. You do not need else and you have to increment total always.